dispatcher.go

```diff
@@ -98,11 +98,32 @@ func NewDispatcher(cfg Config) Dispatcher {
 		panic("a service name is required")
 	}
 
+	// Collect all unique transports from inbounds and outbounds.
```

Comment:
Let's factor this block (101-120) into a seperate function that get's called from here.

Perhaps `func collectUniqueTransports(inbounds, outbounds) []transport.Transport`.